cirq/ops/controlled_operation.py

```diff
@@ -119,8 +119,22 @@ def _resolve_parameters_(self, resolver):
         new_sub_op = protocols.resolve_parameters(self.sub_operation, resolver)
         return ControlledOperation(self.controls, new_sub_op)
 
-    def _trace_distance_bound_(self) -> float:
-        return protocols.trace_distance_bound(self.sub_operation)
+    def _trace_distance_bound_(self):
+        if self._is_parameterized_():
+            return 1
+        if protocols.has_unitary(self.sub_operation):
+            angles = np.sort(
+                np.append(
+                    np.angle(
+                        np.linalg.eigvals(protocols.unitary(
+                            self.sub_operation))), 0))
+            maxim = 2 * np.pi + angles[0] - angles[-1]
+            for i in range(1, len(angles)):
+                maxim = max(maxim, angles[i] - angles[i - 1])
+            if maxim <= np.pi:
+                return 1
+            return np.sin(0.5 * maxim)
```

Comment:
This might need to be a helper function to minimize code copying (maybe `trace_distance_from_eigen_shifts`).

I'm writing a class similar to ControlledOperation where I would also have to copy-paste this.

Reply:
I went with a utility method in `trace_distance_bound` that the relevant files import.